### ISSUE TITLE
R reshape2 lib fails to load (stringi.so / libicui18n.so.64 not found)

### ISSUE THREAD

**Issue by emthomas** [jupyter/docker-stacks#927]
Hi! Thanks for using the Jupyter Docker Stacks.

Please review the following guidance about how to ask questions, contribute changes, or report bugs in the Docker images maintained here.

1. If you have a general question about using these Docker images or with Jupyter + Docker in general, please post your question in the [Jupyter Discourse Q&A section](https://discourse.jupyter.org/c/questions) so that it can be seen, discussed, and hopefully answered by the community.

2. If you are looking to contribute to the images, please see the [Contributor's Guide](http://jupyter-docker-stacks.readthedocs.io/en/latest/#) in the documentation for our preferred processes.

3. If you are reporting an issue with one of the existing images, please answer the questions below to help us troubleshoot the problem. Please be as thorough as possible.

**What docker image you are using?**

`jupyter/all-spark-notebook:2ce7c06a61a1`

**What complete docker command do you run to launch the container (omitting sensitive values)?**

`docker run -it jupyter/all-spark-notebook:2ce7c06a61a1 /bin/bash`

**What steps do you take once the container is running to reproduce the issue?**

```bash
jovyan@484398e1fe17:~$ R
> library(reshape2)
```

**What do you expect to happen?**

Package to load without error

**What actually happens?**

Throws the following error:

```
Error: package or namespace load failed for ‘reshape2’ in dyn.load(file, DLLpath = DLLpath, ...):
 unable to load shared object '/opt/conda/lib/R/library/stringi/libs/stringi.so':
  libicui18n.so.64: cannot open shared object file: No such file or directory
```

**Comment by emthomas** [comment]
Similar to #497. Hope it will be a quick fix.

**Comment by emthomas** [comment]
Workaround is to install stringi. `conda install -c r r-stringi`

**Comment by playermanny2** [comment]
Also having this issue, @emthomas  workaround solves it...any ideas here on a longterm fix@parente  @minrk

**Comment by parente** [comment]
If this is still an issue, the upstream conda-forge `r-reshape2` package definition should be updated to require `r-stringi` if it is indeed a dependency. Then the stacks here should pick up that new `r-reshape2` package version for the fix.

**Comment by matthewbegun** [comment]
Not sure if I should put this here or start a new issue - I get the same error trying to load `tidyverse`. Happy to open a new issue or raise upstream, but it does seem unlikely they would have their dependencies wrong?

**Comment by terzieff** [comment]
Also happens for me, probably linked to a release/update from the last ~7 days, since it was okay last Friday. Using `jupyter/r-notebook:latest` image.

```
Error: package or namespace load failed for ‘tidyverse’ in dyn.load(file, DLLpath = DLLpath, ...):
 unable to load shared object '/opt/conda/lib/R/library/stringi/libs/stringi.so':
  libicui18n.so.64: cannot open shared object file: No such file or directory
```

**Comment by parente** [comment]
In the conda-forge recipes:

* `r-reshape2` lists `r-stringr` as a dependency: https://github.com/conda-forge/r-reshape2-feedstock/blob/master/recipe/meta.yaml#L45
* `r-stringr` lists `r-stringi` as a dependency: https://github.com/conda-forge/r-stringr-feedstock/blob/master/recipe/meta.yaml#L39
* `r-stringi` package is currently broken, missing the same .so listed above: https://github.com/conda-forge/r-stringi-feedstock/issues/20

Fixing the issue in the third bullet should resolve the problem in the images here.

**Comment by dmpe** [comment]
Alternatively, much more prefered, would be to include stringi in all images. That would also help people like me behind firewall :)

**Comment by romainx** [comment]
Hi,

The R stack has been updated recently in the PR #1000 and the library can be loaded.
My guess is that there was an incompatibility between the latest system lib and the previous R version `3.6.1`, now the `3.6.2` is deployed.

```bash
> library(reshape2)
> packageVersion("reshape2")
[1] ‘1.4.3’
```

So could you please give it a try?

However regarding the `reshape2` case it **is now deprecated** and you should switch to `tidyr` (that is also available in the image).

> reshape2 is retired: only changes necessary to keep it on CRAN will be made. We recommend using tidyr instead.
> -- <cite>https://github.com/hadley/reshape</cite>

The most recent version available `1.4.3` last from 2017.



## Side note
We cannot remove the dependency since it is brought by `ggplot2`.
However, we should explicitly install the `tidyverse` package since I think the `tidyr` dependency is brought by `sparklyr`.

**Comment by romainx** [comment]
Hello,

I've no feedback on this issue. Moreover now tests are in place ensuring that all the installed packages can be imported. For example this is an extract of the build logs showing the result of the import test for 
`reshape2`.

```bash
2020-03-17 19:49:47 [    INFO] Grabing the list of specifications ... (helpers.py:76)
-------------------------------- live log call ---------------------------------
2020-03-17 19:49:48 [    INFO] Testing the import of packages ... (test_packages.py:133)
2020-03-17 19:49:48 [    INFO] Trying to import reshape2 (test_packages.py:135)
[...]
PASSED                                                                   [100%]
```
So I'm closing this issue, feel free to open a new one in case of problem.

Best.
